cosmos/signer.go

```diff
@@ -74,61 +77,104 @@ func (x *BurnSignerRunner) UpdateBlocks() {
 func (x *BurnSignerRunner) ValidateInvalidMint(doc *models.InvalidMint) (bool, error) {
 	log.Debug("[BURN SIGNER] Validating invalid mint: ", doc.TransactionHash)
 
-	tx, err := x.poktClient.GetTx(doc.TransactionHash)
+	txResponse, err := x.poktClient.GetTx(doc.TransactionHash)
 	if err != nil {
 		return false, errors.New("Error fetching transaction: " + err.Error())
 	}
 
-	if tx.Tx == "" || tx.TxResult.Code != 0 {
-		log.Debug("[BURN SIGNER] Transaction not found or failed")
-		return false, nil
+	if txResponse == nil {
+		return false, errors.New("transaction not found")
 	}
+	result := util.ValidateTxToCosmosMultisig(txResponse, app.Config.Pocket, uint64(x.poktHeight), x.minimumAmount, x.maximumAmount)
 
-	if tx.TxResult.MessageType != "send" || tx.StdTx.Msg.Type != "pos/Send" {
-		log.Debug("[BURN SIGNER] Transaction message type is not send")
+	if result.TxStatus == models.TransactionStatusFailed {
+		log.Debug("[BURN SIGNER] Invalid Mint Transaction is failed")
 		return false, nil
 	}
 
-	if strings.EqualFold(tx.StdTx.Msg.Value.ToAddress, "0000000000000000000000000000000000000000") {
-		log.Debug("[BURN SIGNER] Transaction recipient is zero address")
+	// NOTE: If mint is disabled, we refund all txs
+	if !app.Config.Pocket.MintDisabled && !result.NeedsRefund {
+		log.Debug("[BURN SIGNER] Invalid Mint Transaction does not need refund")
 		return false, nil
 	}
 
-	if !strings.EqualFold(tx.StdTx.Msg.Value.ToAddress, x.vaultAddress) {
-		log.Debug("[BURN SIGNER] Transaction recipient is not vault address")
-		return false, nil
+	log.Debug("[BURN SIGNER] Validated invalid mint")
+	return true, nil
+}
+
+func (x *BurnSignerRunner) FindMaxSequence() (uint64, error) {
+	lockID, err := LockReadSequences()
+	if err != nil {
+		return 0, fmt.Errorf("could not lock sequences: %w", err)
 	}
+	//nolint:errcheck
+	defer app.DB.Unlock(lockID)
 
-	if !strings.EqualFold(tx.StdTx.Msg.Value.FromAddress, doc.SenderAddress) {
-		log.Debug("[BURN SIGNER] Transaction signer is not sender address")
-		return false, nil
```

Comment:
This condition was missed during validation!

Reply:
Thank you! will add it back